### PR TITLE
try not to throw so many exceptions from Supervision

### PR DESCRIPTION
**Pull request by jsteemann**

### CODE REVIEW

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/1939/

File: arangod/Agency/Node.cpp [hunk 1]
```diff
@@ -947,16 +954,16 @@ std::pair<Node const &, bool> Node::hasAsNode(
...
 
   // *this is bogus initializer
-  std::pair<Node &, bool> fail_pair= {*this, false};
+  std::pair<Node&, bool> fail_pair = {*this, false};
 
   // retrieve node, throws if does not exist
   try {
     Node & target(operator()(url));
-    std::pair<Node &, bool> good_pair={target, true};
+    std::pair<Node &, bool> good_pair = {target, true};
```
Comment (kvahed) [inline arangod/Agency/Node.cpp]: Node&

File: arangod/Agency/Node.cpp [hunk 2]
```diff
@@ -1090,13 +1103,13 @@ std::pair<Node::Children, bool> Node::hasAsChildren(
 } // hasAsChildren
 
 
-std::pair<void *, bool> Node::hasAsBuilder(
-  std::string const & url, Builder & builder, bool showHidden) const {
-  std::pair<void *, bool> ret_pair(nullptr, false);
+std::pair<void*, bool> Node::hasAsBuilder(
+  std::string const& url, Builder &builder, bool showHidden) const {
```
Comment (kvahed) [inline arangod/Agency/Node.cpp]: Builder&

File: arangod/Agency/Node.cpp (hunk 1 above)
Comment (kvahed) [inline arangod/Agency/Node.cpp]: Node&

**Pull request merged**